dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/SqlViewController.java

```diff
@@ -283,4 +274,109 @@ private GridResponse buildResponse( SqlView sqlView, SqlViewQuery query )
         }
         return new GridResponse( query.isSkipPaging() ? null : query.getPager(), grid );
     }
+
+    @Override
+    protected void preCreateEntity( SqlView entity )
+        throws ConflictException
+    {
+        checkJobConfigurationExists( entity );
+    }
+
+    @Override
+    protected void postCreateEntity( SqlView entity )
+    {
+        if ( entity.getUpdateJobId() != null )
+        {
+            addAndUpdateIfNeeded( entity );
+        }
+    }
+
+    @Override
+    protected void preUpdateEntity( SqlView entity, SqlView newEntity )
+        throws ConflictException
+    {
+        removeAndUpdateIfNeeded( newEntity );
+        if ( newEntity.getUpdateJobId() != null )
+        {
+            checkJobConfigurationExists( newEntity ); // validate
+            addAndUpdateIfNeeded( newEntity );
+        }
+    }
+
+    @Override
+    protected void prePatchEntity( SqlView entity, SqlView newEntity )
+        throws ConflictException
+    {
+        preUpdateEntity( entity, newEntity );
+    }
+
+    @Override
+    protected void preDeleteEntity( SqlView entity )
+    {
+        removeAndUpdateIfNeeded( entity );
+    }
+
+    private void checkJobConfigurationExists( SqlView entity )
+        throws ConflictException
+    {
+        String jobId = entity.getUpdateJobId();
+        if ( jobId == null )
+        {
+            return;
+        }
+        JobConfiguration config = jobConfigurationService.getJobConfigurationByUid( jobId );
+        if ( config == null )
+        {
+            throw new ConflictException( "Update job does not exist: " + jobId );
+        }
+        if ( config.getJobType() != JobType.SQL_VIEW_UPDATE )
+        {
+            throw new ConflictException(
+                format( "Update job must be of type %s but was: %s", JobType.SQL_VIEW_UPDATE, config.getJobType() ) );
+        }
+    }
+
+    private void addAndUpdateIfNeeded( SqlView entity )
+    {
+        JobConfiguration config = jobConfigurationService.getJobConfigurationByUid( entity.getUpdateJobId() );
+        if ( config == null )
+        {
+            return;
+        }
+        List<String> sqlViews = getSqlViews( config );
+        if ( !sqlViews.contains( entity.getUid() ) )
+        {
+            sqlViews.add( entity.getUid() );
+            jobConfigurationService.updateJobConfiguration( config );
+        }
+    }
+
+    private void removeAndUpdateIfNeeded( SqlView entity )
```

Comment:
This part _feels_ a bit complex to sit in a controller, perhaps some of this could be moved to service layer and tested - I could be wrong.